### ISSUE TITLE
Document remap for both Injections and At usages

### ISSUE THREAD

**Issue by gabizou** [SpongePowered/Mixin#118]
In a recent code cleanup session with Sponge, I found that many injections that were targeting methods that were mapped to classes like `java.util.List`, `java.util.UUID` etc. would fail to be remapped (for obvious reasons) and the Mixin AP would output a warning. I found that this wasn't really well documented in the wiki, so I'm suggesting that it is documented and made aware for mods alike to correctly mark those injections as `remap = false`.

I only found this out due to the warnings output from Mixin's AP, but still, it's worth documenting.

**Comment by Mumfrey** [comment]
Yeah I've observed that `remap` is one of the least understood arguments, though it is currently documented in [the javadoc](http://jenkins.liteloader.com/job/Mixin/javadoc/index.html?org/spongepowered/asm/mixin/injection/At.html) (as are most things) though I will make a note to include it in the AP documentation when I get to that.
